app/controllers/assignments_controller.rb

```diff
@@ -450,7 +450,8 @@ def populate_starter_file_manager
              sections: section_data,
              available_after_due: assignment.starter_files_after_due,
              starterfileType: assignment.starter_file_type,
-             defaultStarterFileGroup: assignment.default_starter_file_group&.id || '' }
+             defaultStarterFileGroup: assignment.default_starter_file_group&.id || '',
+             readOnly: current_role.ta? } # FIXME: change to be NOT authorized graders instead of all TAs
```

Comment:
How would we make this more general instead of restricting for _all_ TAs? Ideally, I'd want the result of the `manage?` method in `app/policies/ApplicationPolicy.rb`, but I can't (and shouldn't) get that here, right?

Reply:
EDIT: this also works (from tracing through the `manage?` method and internal method calls):  `(current_role.instructor? || current_role.grader_permission.manage_submissions)`